### DISCUSSION TITLE
Will mods ever be made universal? (not exclusive to mac, windows, ect.)

### DISCUSSION THREAD

**Discussion by TheQuestionAsker2** [geode-sdk/sdk#189] [Help] [answered]
A lot of really good/popular mods are exclusive to mac, so I wanted to know if mods would ever be available for download on all devices. it would be amazing and I think it would be really beneficial to have.

**Comment by HJfod** [comment, marked as answer]
This is something we can't guarantee automatically, as the mod has to be separately compiled for each platform and may depend on platform-specific code that has to be ported.

However, Geode has been designed with this in mind, so all of its APIs are cross-platform and mods can contain binaries for different platforms. The intent is that for most mods, if the developer uses intended Geode tools they can simply use a GitHub Action to build the binaries for other platforms. Examples of mods that already do this are GDShare and BetterEdit.

If we add scripting language support in the future (which I'd definitely like to see) those will likely be cross-platform off-the-bat.

TL;DR; It's up to the developer of the mod, but we try to make it easy for them.
